### PR TITLE
[8.0] porting of ricevute bancarie

### PR DESCRIPTION
**Pull request by scigghia**
here my porting of l10n_it_ricevute_bancarie for Odoo

### CODE REVIEW

**Comment by giudalo** [comment]
:+1:

**Comment by LeartS** [comment]
Uhm, half new API and half old?

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import fields, models, api, _, exceptions, workflow
+import time
+import openerp.addons.decimal_precision as dp
```
Comment (LeartS) [inline l10n_it_ricevute_bancarie/models/riba.py]: See PEP for import ordering and grouping: https://www.python.org/dev/peps/pep-0008/#imports

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+            res[r[0]].append(r[1])
+        return res
+
+    # return the ids of the move lines which has the same account than the
+    # statement whose id is in ids
+    @api.multi
+    def move_line_id_payment_get(self):  # , cr, uid, ids, *args):
+        if not self:
+            return []
+        ids = [x.id for x in self]
```
Comment (LeartS) [inline l10n_it_ricevute_bancarie/models/riba.py]: Isn't this just self.ids? https://www.odoo.com/documentation/8.0/reference/orm.html#model-reference (section recordset operations)

**Comment by scigghia** [comment]
hum! seems that this PR is a work in progress @LeartS 
if you want to check features, or if you want to report where to improve the code, you are welcome

(i prefer to check functionality, first)

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+        res = self.move_line_id_payment_get(ids)
+        if not res:
+            return False
+        ok = True
+        for id in res:
+            cr.execute(
+                'select reconcile_id from account_move_line where id=%s',
+                (id,))
+            ok = ok and bool(cr.fetchone()[0])
+        return ok
```
Comment (LeartS) [inline l10n_it_ricevute_bancarie/models/riba.py]: To refactor using ORM and single select query

File: l10n_it_ricevute_bancarie/wizard/wizard_riba_issue.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            'date_created': fields.Date.context_today(self),
+        }
+        rd_id = riba_list.create(rd).id
+
+        # group by partner and due date
+        grouped_lines = {}
+        move_lines = move_line_obj.search(
+            [('id', 'in', self._context['active_ids'])])
+        #move_lines = move_line_obj.browse(move_line_ids)
+        #import pdb; pdb.set_trace()
```
Comment (LeartS) [inline l10n_it_ricevute_bancarie/wizard/wizard_riba_issue.py]: Dev leftover

**Comment by LeartS** [comment]
Oh sorry. You linked it in the IRC channel so I assumed you wanted a review.

**Comment by tafaRU** [comment]
@scigghia, is this PR still work in progress? Does it need the related label?

**Comment by scigghia** [comment]
I need someone to test it, from the begin to the end.
for me, it works
obviously it is not entirely ported to new v8 api, we can do that after it works for others, not only for me ;)

**Comment by eLBati** [comment]
As https://github.com/OCA/l10n-italy/pull/15 is merged, it should probably be included in v8 version

**Comment by scigghia** [comment]
:hear_no_evil:

**Comment by scigghia** [comment]
now it is mergeable
:monkey:

**Comment by eLBati** [comment]
```l10n_it_ipa/README.rst``` and ```l10n_it_rea/README.rst``` should not be involved by this PR

**Comment by scigghia** [comment]
now, is ok?

**Comment by eLBati** [comment]
Diff between 7.0 and this
--------------------------------

See [account.py diff between 7.0 and this](https://gist.github.com/eLBati/2f0fe985bc631ecf821e)
 - lines 29 and 41 of diff: ```default``` is unnecessary


[riba.py diff](https://gist.github.com/eLBati/d46052662c03183e0cf8)
 - Please do not remove copyrights
 - As model names ```riba.distinta*``` are changed,  we should probably provide a migration script from 7.0 to 8.0. Otherwise, after migrating to 8.0, every old riba list would be lost. I would keep the old name in this case. What do people think?

[riba_config.py diff](https://gist.github.com/eLBati/6c19e316a7b92593afd6)
 - about ```riba.configurazione``` same question as ```riba.distinta```
 - ```get_default_value_by_list*``` are not not used

Travis
-------

See https://travis-ci.org/OCA/l10n-italy/jobs/66731226 for flake8 and pylint
(it seems some PEP8 changes present in 7.0 has not been included in this PR)

About https://travis-ci.org/OCA/l10n-italy/jobs/66731228, ```account_due_list``` is available at https://github.com/OCA/account-payment so you should just add it as dependency using [oca_dependencies.txt](https://github.com/OCA/maintainer-quality-tools/blob/master/sample_files/oca_dependencies.txt) (simply add ```account-payment``` line to https://github.com/OCA/l10n-italy/blob/8.0/oca_dependencies.txt)

**Comment by scigghia** [comment]
now, i think we must discuss about migration script from 7.0 to 8.0
however , when we talked about name fields changes, no one thinked about (riba) migration

File: l10n_it_ipa/README.rst [hunk 5]
```diff
@@ -1,47 +0,0 @@
...
-
-.. image:: http://odoo-community.org/logo.png
-   :alt: Odoo Community Association
-   :target: http://odoo-community.org
-
-This module is maintained by the OCA.
-
-OCA, or the Odoo Community Association, is a nonprofit organization whose mission is to support the collaborative development of Odoo features and promote its widespread use.
-
-To contribute to this module, please visit http://odoo-community.org.
```
Comment (eLBati) [inline l10n_it_ipa/README.rst]: Please do not remove this file from repo

File: l10n_it_rea/README.rst [hunk 6]
```diff
@@ -1,38 +0,0 @@
...
-.. image:: http://odoo-community.org/logo.png
-   :alt: Odoo Community Association
-   :target: http://odoo-community.org
-
-This module is maintained by the OCA.
-
-OCA, or the Odoo Community Association, is a nonprofit organization whose mission is to support the collaborative development of Odoo features and
-promote its widespread use.
-
-To contribute to this module, please visit http://odoo-community.org.
```
Comment (eLBati) [inline l10n_it_rea/README.rst]: Please do not remove this file from repo

**Comment by eLBati** [comment]
About model name change, in my previous message I was proposing to keep the old names (only for table names thus)

About travis, flake8 and pylint are still failing.

Automated tests are failing for
```
File "/home/travis/odoo-8.0/openerp/fields.py", line 499, in _setup_related
    field = recs._fields[name]
KeyError: 'iban'
```
Did you check whether this happens in your local instance?

About l10n_it_ipa/README.rst and l10n_it_rea/README.rst, they result chenged again https://github.com/OCA/l10n-italy/pull/83/files#diff-157ec115fe5dfb0ed6dfcc6f785742c6
I suggest to rebase removing every change that involves those files

More in general, about corrective commits (like https://github.com/OCA/l10n-italy/commit/ebd06fd9b20dbf6998fca744e49a36aeea327654), please consider [amend](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-the-Last-Commit) and [squash](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits) and please always use meaningful commit messages.

Thanks

**Comment by scigghia** [comment]
going to PR https://github.com/OCA/l10n-italy/pull/115

**Pull request closed without merging**